airbyte-integrations/connectors/source-sendgrid/source_sendgrid/config_migrations.py

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from typing import Any, List, Mapping
+
+import pendulum
+from airbyte_cdk.config_observation import create_connector_config_control_message
+from airbyte_cdk.entrypoint import AirbyteEntrypoint
+from airbyte_cdk.sources import Source
+from airbyte_cdk.sources.message import InMemoryMessageRepository, MessageRepository
+
+logger = logging.getLogger("airbyte_logger")
+
+
+class MigrateToLowcodeConfig:
+    """
+    This class stands for migrating the config at runtime.
+    Specifically, starting from `0.5.0`, apikey moved to api_key and start_time moved to start_date.
+    start_time was not required and not start_date is.
+    """
+
+    message_repository: MessageRepository = InMemoryMessageRepository()
+
+    @classmethod
+    def should_migrate(cls, config: Mapping[str, Any]) -> bool:
+        """
+        This method determines if the config should be migrated.
+        Returns:
+            > True, if the transformation is necessary
+            > False, otherwise.
+        """
+        has_api = "apikey" in config
+        has_time = "start_time" in config
+        return has_api or has_time
+
+    @staticmethod
+    def transform_config(config: Mapping[str, Any]) -> Mapping[str, Any]:
+        # move api_secret inside credentials block
+        if "apikey" in config:
+            config["api_key"] = config["apikey"]
+            config.pop("apikey")
```

Comment:
You should not delete the previous parameters. This will make the config backwards compatible, so old parameters will be used when reverting to the earlier versions. Unfortunately, this will only work for the users who created connections before this migration; new users would need to input parameters again when reverting to the versions previous to yours.

Reply:
So you are saying I should not `pop` this here? Do they go in the spec too?

Reply:
Yes, you should remove pop. Old parameters are used only when reverting to the previous version, so they don't need to be in the new spec.

Reply:
Ok, done. But note, I think just removing `pop` is risky. I don't want the mnigration to run every time, so I added logic to have it check for the new keys. Most other migrations I've seen only check for the old keys.

```
        has_api = "apikey" in config
        has_time = "start_time" in config
        has_newapi = "api_key" in config
        has_newdate = "start_date" in config
        if has_newapi and has_newdate:
            return False
        return has_api or has_time
```